drivers/sensor/bh1749/bh1749.h

```diff
@@ -0,0 +1,125 @@
+#define BH1749_MODE_CONTROL1_MEAS_MODE_120MS            0x02
+#define BH1749_MODE_CONTROL1_MEAS_MODE_240MS            0x03
+#define BH1749_MODE_CONTROL1_MEAS_MODE_35MS             0x05
+
+#define BH1749_MODE_CONTROL1_RGB_GAIN_Msk               GENMASK(4, 3)
+#define BH1749_MODE_CONTROL1_RGB_GAIN_1X                ((0x01) << 3)
+#define BH1749_MODE_CONTROL1_RGB_GAIN_32X               ((0x03) << 3)
+
+#define BH1749_MODE_CONTROL1_IR_GAIN_Msk                GENMASK(6, 5)
+#define BH1749_MODE_CONTROL1_IR_GAIN_1X                 ((0x01) << 5)
+#define BH1749_MODE_CONTROL1_IR_GAIN_32X                ((0x03) << 5)
+
+/* BH1749_MODE_CONTROL2 */
+#define BH1749_MODE_CONTROL2_RGB_EN_Msk                 BIT(4)
+#define BH1749_MODE_CONTROL2_RGB_EN_ENABLE              BIT(4)
+#define BH1749_MODE_CONTROL2_RGB_EN_DISABLE             0x00
+
+#define BH1749_MODE_CONTROL2_VALID_Msk                  BIT(7)
+
+/* BH1749_INTERRUPT */
+#define BH1749_INTERRUPT_ENABLE_Msk                     BIT(0)
+#define BH1749_INTERRUPT_ENABLE_DISABLE                 0x00
+#define BH1749_INTERRUPT_ENABLE_ENABLE                  BIT(0)
+
+#define BH1749_INTERRUPT_LATCH                          BIT(4)
+
+#define BH1749_INTERRUPT_INT_SOURCE_Msk                 GENMASK(3, 2)
+#define BH1749_INTERRUPT_INT_SOURCE_RED                 ((0x00) << 2)
+#define BH1749_INTERRUPT_INT_SOURCE_GREEN               ((0x01) << 2)
+#define BH1749_INTERRUPT_INT_SOURCE_BLUE                ((0x02) << 2)
+
+#define BH1749_INTERRUPT_INT_STATUS_Msk                 BIT(7)
+
+/* BH1749_PERSISTENCE */
+#define BH1749_PERSISTENCE_PERSISTENCE_Msk              GENMASK(1, 0)
+#define BH1749_PERSISTENCE_PERSISTENCE_ACTIVE_END       0x00
+#define BH1749_PERSISTENCE_PERSISTENCE_UPDATE_END       0x01
+#define BH1749_PERSISTENCE_PERSISTENCE_4_SAMPLES        0x02
+#define BH1749_PERSISTENCE_PERSISTENCE_8_SAMPLES        0x03
+
+/* BH1749 RGB/IR SAMPLE POSITIONS  */
+#define BH1749_RGB_BYTE_POS_RED                         0
+#define BH1749_RGB_BYTE_POS_GREEN                       1
+#define BH1749_RGB_BYTE_POS_BLUE                        2
+#define BH1749_RGB_BYTE_POS_IR                          4
+
+/* Manufacturer ID */
+#define BH1749_MANUFACTURER_ID_DEFAULT                  0xE0
+
+struct bh1749_data {
+	struct device *i2c;
+	struct device *gpio;
+	struct gpio_callback gpio_cb;
+	struct k_work work;
+	struct device *dev;
+	u16_t sample_rgb_ir[5];
+	u8_t pdata;
+
+#ifdef CONFIG_BH1749_TRIGGER
+	sensor_trigger_handler_t trg_handler;
+	struct sensor_trigger trigger;
+	struct k_sem data_sem;
+#endif
+};
+
+#ifdef CONFIG_BH1749_TRIGGER
+void bh1749_work_cb(struct k_work *work);
+
+int bh1749_attr_set(struct device *dev,
+		    enum sensor_channel chan,
+		    enum sensor_attribute attr,
+		    const struct sensor_value *val);
+
+int bh1749_trigger_set(struct device *dev,
+		       const struct sensor_trigger *trig,
+		       sensor_trigger_handler_t handler);
+
+int bh1749_gpio_interrupt_init(struct device *dev);
```

Comment:
I don't entirely get the idea of splitting the sensor code into two file. Wouldn't it be easier to have all this stuff in one file and avoid this declarations? You can still cut out the trigger related functions with when `BH1749_TRIGGER` is not enabled.

Reply:
I get that, but if we look at the zephyr source tree, drivers\sensor\*, we see that this is common practice, and I don't want to move away from that.

Reply:
Do you agree with keeping the file structure?

Reply:
I'm fine with that. I still don't get what are the benefits of such split (if there are any, I see only downsides). Perhaps it was needed for some reason in a driver, and was just copied in others added later? Anyway, I won't insist on changing it, it's not a big issue, so let's keep it as it is.